Refresher/UI/PatchForm.cs

```diff
@@ -0,0 +1,168 @@
+    {
+        MessageBox.Show("No guide exists yet, stay tuned!", MessageBoxType.Warning);
+    }
+
+    private void Patch(object? sender, EventArgs e)
+    {
+        if (!this._patchButton.Enabled) return; // shouldn't happen ever but just in-case
+        if (this._patcher == null) return;
+        if (this._tempFile == null) return;
+        
+        this._patcher.PatchUrl(this._urlField.Text);
+
+        // Warn user if file already exists
+        // Technically most file pickers already handle this for us but still
+        DialogResult result = DialogResult.Yes;
+        if (File.Exists(this._outputFileField.FilePath))
+        {
+            result = MessageBox.Show("You are overwriting an existing EBOOT. Are you sure you want to do this?",
+                MessageBoxButtons.YesNo, MessageBoxType.Warning);
+        }
+
+        if (result == DialogResult.Yes)
+        {
+            File.Move(this._tempFile, this._outputFileField.FilePath, true);
+            MessageBox.Show("Successfully patched EBOOT!");
+        }
+
+        // Re-initializes patcher so we can patch with the same parameters again
+        // Probably slow but prevents crash
+        this.FileUpdated(this, EventArgs.Empty);
+    }
+
+    private void FailVerify(string reason, bool clear = true)
+    {
+        if(clear) this._problems.Items.Clear();
+        this._problems.Items.Add(reason);
+        
+        try
+        {
+            if (this._tempFile != null) File.Delete(this._tempFile);
+        }
+        catch
+        {
+            // ignored
+        }
+        
+        this._patcher = null;
+        this._patchButton.Enabled = false;
+        this._mappedFile?.Dispose();
+    }
+
+    private void FileUpdated(object? sender, EventArgs ev)
+    {
+        try
+        {
+            //Create a temp file to store the EBOOT as we work on it
+            this._tempFile = Path.GetTempFileName();
+
+            //Copy the input file to the temp file
+            File.Copy(this._inputFileField.FilePath, this._tempFile, true);
+        }
+        catch (Exception e)
+        {
+            this.FailVerify("Could not create and copy to temporary file.\n" + e);
+            return;
+        }
+
+        try
+        {
+            this._mappedFile?.Dispose();
+            this._mappedFile = MemoryMappedFile.CreateFromFile(this._tempFile, FileMode.Open, null, 0, MemoryMappedFileAccess.ReadWrite);
+            this._patcher = new Patcher(this._mappedFile.CreateViewStream());
+        }
+        catch(Exception e)
+        {
+            this.FailVerify("Could not read data from the input file.\n" + e);
+            return;
+        }
+
+        this.FormUpdated(sender, ev);
+    }
+
+    private void FormUpdated(object? sender, EventArgs e)
+    {
+        if (this._patcher == null) return;
+        
+        this._problems.Items.Clear();
+        List<Message> messages = this._patcher.Verify(this._urlField.Text).ToList();
```

Comment:
This is slow when used synchronously on a UI thread (half a second is definitely noticable and makes the UI feel very sluggist), and should be made asynchronous

Reply:
Doesnt Eto have like an App.Invoke thing? you can just shove it onto another thread, then use App.Invoke at the very end to ensure thread safety

Reply:
But yes, its fine to postpone to after this PR